vss-core/src/main/java/org/eclipse/kuksa/vsscore/model/VssSpecification.kt

```diff
@@ -106,21 +106,16 @@ val VssSpecification.parentKey: String
 /**
  * Iterates through all nested children which also may have children and aggregates them into one big collection.
  */
-val VssSpecification.heritage: List<VssSpecification>
+val VssSpecification.heritage: Collection<VssSpecification>
     get() = children.toList() + children.flatMap { it.heritage }
 
 /**
- * Creates an inheritance line to the given heir. Similar to [vssPathHeritageLine] but the other way around.
- *
- * @param heir where the inheritance line should stop
- * @return a [Collection] of the full heritage line in the form of [VssSpecification]
+ * Finds the latest generation in the form of [VssProperty] for the current [VssSpecification].
  */
-fun VssSpecification.findHeritageLine(heir: VssSpecification): List<VssSpecification> {
-    val specificationKeys = heir.vssPathHeritageLine
-    return heritage.filter { child ->
-        specificationKeys.contains(child.vssPath)
-    }
-}
+val VssSpecification.latestGeneration: Collection<VssProperty<*>>
```

Comment:
This sounds like a movie title: "Star Trek - The Latest Generation" but jokes aside, I have absolutely no clue, when I would (have to) use this latestGeneration and what I should expect from this

Reply:
Whenever you want get all available VssProperties (youngest generation without children). I could rename it to just vssProperties if that helps :)

Reply:
Yeah, I think renaming it to vssProperties would make more sense if it really reflects all vssProperties of a VssSpecification